File: backend/main/test/groupher_server_web/controller/health_test.exs
defmodule GroupherServerWeb.Test.Controller.Health do
  @moduledoc false

  use GroupherServer.TestMate

  test "returns the shared health contract response" do
    conn = build_conn()

    res =
      conn
      |> get("/health")
      |> json_response(200)

    assert res["schemaVersion"] == "health.v1"
    assert res["status"] == "ok"
    assert res["service"] == "phoenix"
    assert is_binary(res["version"])
    assert is_binary(res["environment"])
    assert is_binary(res["timestamp"])
    assert is_integer(res["uptimeMs"])
    assert res["checks"] == []
  end
end

File: frontend/community/src/server/health.test.ts
import { SERVICE_HEALTH_SERVICES } from '@groupher/contracts/health'
import { describe, expect, it } from 'vitest'

import { buildCommunityHealth } from './health'

describe('Community health.v1', () => {
  it('returns every required contract field with isolate uptime', () => {
    const response = buildCommunityHealth({
      environment: 'test',
      now: new Date('2026-08-22T00:00:00.000Z'),
      uptimeMs: 123,
      version: 'test-sha',
    })

    expect(response).toEqual({
      schemaVersion: 'health.v1',
      status: 'ok',
      service: 'community',
      version: 'test-sha',
      environment: 'test',
      timestamp: '2026-08-22T00:00:00.000Z',
      uptimeMs: 123,
      checks: [],
    })
    expect(SERVICE_HEALTH_SERVICES).toContain('community')
  })
})
